deepchecks/tabular/checks/model_evaluation/train_test_performance.py

```diff
@@ -112,9 +112,13 @@ def run_logic(self, context: Context) -> CheckResult:
         results_df = pd.DataFrame(results, columns=['Dataset', 'Class', 'Metric', 'Value', 'Number of samples'])
 
         if context.with_display:
+            results_df_for_display = results_df.copy()
+            results_df_for_display['Dataset']\
+                .replace({DEFAULT_DATASET_NAMES[0]: train_dataset.name, DEFAULT_DATASET_NAMES[1]: test_dataset.name},
```

Comment:
@noamzbr If I change it at line 96 it will also change the CheckResult.value, and not just the display

Reply:
I see there is a question here of modifying the structure of the .value, because conditions use it and users may want to parse it in a coherent manner. 
@ItayGabbay WDYT?

Reply:
Also, want to hear what @nirhutnik thinks because in the issue he specifically mentioned not to do it